src/coreclr/vm/arm/cgencpu.h

```diff
@@ -99,11 +99,14 @@ EXTERN_C void setFPReturn(int fpSize, INT64 retVal);
 // Parameter size
 //**********************************************************************
 
-typedef INT32 StackElemType;
-#define STACK_ELEM_SIZE sizeof(StackElemType)
-
-// !! This expression assumes STACK_ELEM_SIZE is a power of 2.
-#define StackElemSize(parmSize) (((parmSize) + STACK_ELEM_SIZE - 1) & ~((ULONG)(STACK_ELEM_SIZE - 1)))
+inline unsigned StackElemSize(unsigned parmSize, bool isValueType = false /* unused */, bool isFloatHfa = false /* unused */)
+{
+    typedef INT32 StackElemType;
+    const unsigned stackSlotSize = sizeof(StackElemType);
+    // The next expression assumes STACK_ELEM_SIZE is a power of 2.
+    static_assert(((stackSlotSize & (stackSlotSize-1)) == 0), "STACK_ELEM_SIZE must be a power of 2");
```

Comment:
It seems that the static assert is useless when the constant and type is defined in the same function. And since you've changed all the code to use TARGET_POINTER_SIZE instead of STACK_ELEMENT_SIZE, let's use it here as well instead of the stackSlotSize.

Reply:
`TARGET_POINTER_SIZE` is defined here https://github.com/dotnet/runtime/blob/f765507f226f4c0b6b27601769cc9b1e09468935/src/coreclr/inc/gcinfotypes.h#L608 and not available for cgencpu.h, should I redefine them or include gcinfotypes.h?

Reply:
Hmm, it is not worth including that stuff here. I actually think that there is too much ceremony in this function (not created by you, but stemming from the previous implementation. It seems that just using sizeof(void*) instead of typedefing StackElemType as INT32 or INT64 and then taking sizeof that one and then static asserting that it is a power of two would be clear. There are hundreds of places in the code base that use sizeof(void*) as a stack slot size anyways.